main.nf

```diff
@@ -237,71 +237,143 @@ process QC_Denoise_T1 {
-    .collect(sort:true)
-    .into{b0_corrected;for_counter_b0_corrected}
+    .fromPath(["$input/**/Eddy_Topup/*bvec", "$input/**/Eddy/*bvec"])
+      .collect(sort:true)
+      .set{bvecs_for_eddy_topup}
 
-for_counter_b0
-    .flatten()
-    .count()
-    .set{counter_b0}
+Channel
+      .fromPath(["$input/**/Eddy_Topup/*_corrected.nii.gz", "$input/**/Eddy/*_corrected.nii.gz"])
+      .collect(sort:true)
+      .set{dwi_corrected_for_eddy_topup}
 
-for_counter_b0_corrected
-    .flatten()
-    .count()
-    .set{counter_b0_eddy}
+Channel
+      .fromPath("$input/**/Eddy_Topup/*_mask.nii.gz")
+      .collect(sort:true)
+      .set{mask_for_eddy_topup}
+
+Channel
+      .fromPath(["$input/**/Eddy_Topup/*bval_eddy", "$input/**/Eddy/*bval_eddy"])
+        .collect(sort:true)
+        .set{bvals_for_eddy_topup}
 
 process QC_Eddy_Topup {
     cpus params.eddy_topup_nb_threads
 
     input:
-    file(b0) from b0_for_eddy_topup
-    file(b0_corrected) from b0_corrected
-    file(mask) from b0_mask_for_eddy_topup
-    val(counter_b0_before) from counter_b0
-    val(counter_b0_corrected) from counter_b0_eddy
+    file(dwi_before) from dwi_before_for_eddy_topup
+    file(dwi_after) from dwi_corrected_for_eddy_topup
+    file(bvals) from bvals_for_eddy_topup
+    file(bvecs) from bvecs_for_eddy_topup
 
     output:
     file "report_eddy_topup.html"
     file "data"
     file "libs"
+    files "images/*gz"
 
     when:
-    (counter_b0_before == counter_b0_corrected) && params.run_qc_eddy_topup
+    params.run_qc_eddy_topup
 
     script:
     """
     export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
     export OMP_NUM_THREADS=1
     export OPENBLAS_NUM_THREADS=1
-    for i in $b0
+    for i in $dwi_before
+    do
+        echo \$i >> dwi_before.txt
+    done
+    for i in $dwi_after
     do
-        echo \$i >> b0.txt
+        echo \$i >> dwi_after.txt
     done
-    for i in $b0_corrected
+    for i in $bvals
     do
-        echo \$i >> b0_corrected.txt
+        echo \$i >> bvals.txt
     done
-    for i in $mask
+    for i in $bvecs
     do
-        echo \$i >> mask.txt
+        echo \$i >> bvecs.txt
     done
     mkdir images
-    paste b0.txt b0_corrected.txt mask.txt | while read a b c; do filename=\$(basename -- "\$b");\
-    filename="\${filename%.*.*}"; mrcalc \$b \$c -mult \${filename}_corrected_masked.nii.gz;\
-    mrcat \$a \${filename}_corrected_masked.nii.gz images/\${filename}_eddy_topup.nii.gz; done
+    paste dwi_before.txt dwi_after.txt bvals.txt bvecs.txt | while read a b c d;
+    do
+    filename=\$(basename -- "\$b");\
+    filename="\${filename%.*.*}";
+    scil_extract_b0.py --mean --b0_thr 50 \$a \$c \$d \${filename}_b0_before.nii.gz;\
+    scil_extract_b0.py --mean --b0_thr 50 \$b \$c \$d \${filename}_b0_after.nii.gz;\
+    mrcat \${filename}_b0_before.nii.gz \${filename}_b0_after.nii.gz images/\${filename}__eddy_topup.nii.gz
+    done
 
     dmriqc_generic.py "Eddy Topup" report_eddy_topup.html\
```

Comment:
This must be an error with dmriqcpy, the Gif numbering is bad. It goes from 0 to 1 instead of 1 to 2. In any case, I think a new script to do before/after Gifs is required for this kind of thing, numbers don't mean anything, we'd have to make before/after labels in the images. To put on our TODO list.